maskrcnn_benchmark/engine/trainer.py

```diff
@@ -72,6 +73,11 @@ def do_train(
         losses_reduced = sum(loss for loss in loss_dict_reduced.values())
         meters.update(loss=losses_reduced, **loss_dict_reduced)
 
+        if tb_logger:
```

Comment:
What about adding support for TensorBoard directly in the [`MetricLogger`](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/maskrcnn_benchmark/utils/metric_logger.py)?
We could pass the `meters` as an argument to `do_train`, and it can be setup to use TensorBoard or not in the `train_net` file.

I was thinking about something like
```python
class MetricLogger(object):
    def __init__(self, delimiter="\t", summary_writer=None):
        self.meters = defaultdict(SmoothedValue)
        self.delimiter = delimiter
        self.summary_writer = summary_writer

    def update(self, **kwargs):
        for k, v in kwargs.items():
            if isinstance(v, torch.Tensor):
                v = v.item()
            assert isinstance(v, (float, int))
            self.meters[k].update(v)
             if self.summary_writer:
                 tb_logger.add_scalar(k, v, iteration)
...
```

We would need to have a reliable way of getting the `iteration`, but that shouldn't be too bad / hard.
This way we don't need to change almost anything in the training code.

What do you think?

Reply:
iteration doesn't exist there.

would this be okay for you then?

```python
class MetricLogger(object):
    def __init__(self, delimiter="\t", summary_writer=None):
        self.meters = defaultdict(SmoothedValue)
        self.delimiter = delimiter
        self.summary_writer = summary_writer

    def update(self, **kwargs):
        for k, v in kwargs.items():
            if isinstance(v, torch.Tensor):
                v = v.item()
            assert isinstance(v, (float, int))
            self.meters[k].update(v)
            if self.summary_writer:
                self.summary_writer.add_scalar(k, v, self.meters[k].count)
```

Reply:
I actually think that I'd prefer another class for the TFLogger, as I mentioned before.

About the `.count`, this will work fine as long as the user doesn't want to restart training, in which case the iteration will be off.